OptimizelySDK/Utils/ConditionTreeEvaluator.cs

```diff
@@ -50,7 +45,7 @@ public class ConditionEvaluator
         /// <param name="userAttributes">Hash representing user attributes</param>
         /// <returns>true/false if the user attributes match/don't match the given conditions,
         /// null if the user attributes and conditions can't be evaluated</returns>
-        private bool? AndEvaluator(JArray conditions, UserAttributes userAttributes)
+        private bool? AndEvaluator(JArray conditions, Func<JToken, bool?> leafEvaluator)
```

Comment:
What is this second function arg we are taking? How come we are using the JToken generic? Can you explain what is going on here?

Reply:
The second argument specify return type of callback which, in this case, is nullable bool for leaf evaluator. Also, we use JToken to represent condition object that is passed to leaf evaluator.